WalletWasabi/JsonConverters/ExtPubKeyJsonConverter.cs

```diff
@@ -4,28 +4,25 @@
 
 namespace WalletWasabi.JsonConverters;
 
-public class ExtPubKeyJsonConverter : JsonConverter
+public class ExtPubKeyJsonConverter : JsonConverter<ExtPubKey>
 {
 	/// <inheritdoc />
-	public override bool CanConvert(Type objectType)
+	public override ExtPubKey? ReadJson(JsonReader reader, Type objectType, ExtPubKey? existingValue, bool hasExistingValue, JsonSerializer serializer)
 	{
-		return objectType == typeof(ExtPubKey);
-	}
+		var s = (string?)reader.Value;
+		if (s is not null)
+		{
+			ExtPubKey epk = NBitcoinHelpers.BetterParseExtPubKey(s);
+			return epk;
+		}
 
-	/// <inheritdoc />
-	public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
-	{
-		var s = (string)reader.Value;
-		ExtPubKey epk = NBitcoinHelpers.BetterParseExtPubKey(s);
-		return epk;
+		return null;
```

Comment:
Should this throw here:

```suggestion
		throw new ArgumentNullException(nameof(reader.Value));
```

Reply:
I guess not. Because using `throw new ArgumentNullException(nameof(reader.Value));` in #7006 caused the issue fixed by #7010